lib/article_json/import/google_doc/html/image_parser.rb

```diff
@@ -36,17 +37,42 @@ def float
             super if floatable_size?
           end
 
+          # Extracts an href from the tag [image-link-to: url]) if present
+          # in the caption node.
+          # @return [String]
+          def href
+            return if @caption_node.nil?
+            match = @caption_node.content.strip.match(href_regexp)
+            return if match.nil?
+            remove_image_link_tag
+            match[:url]
+          end
+
           # @return [ArticleJSON::Elements::Image]
           def element
             ArticleJSON::Elements::Image.new(
               source_url: source_url,
               float: float,
-              caption: caption
+              caption: caption,
+              href: @href
             )
           end
 
           private
 
+          # Removes the [image-link-to: url] tag from the caption node
+          def remove_image_link_tag
+            @caption_node
+              .children
+              .first
+              .content = @caption_node.content.gsub(href_regexp, '').strip
+          end
+
+          # Regular expression to chech if there's a [image-link-to: url] tag
+          # @return [Regexp]
+          def href_regexp
+            %r{\[image-link-to: (?<url>.*)\]}
```

Comment:
Oh, just realized that since regexes are by default greedy, right now, `(?<url>.*)` would include everything in the `url` match up until the _last_ closing `]` in the string. Therefore, a caption like:
```ruby
string = '[image-link-to: https://www.devex.com] [no-caption]'
regex = %r{\[image-link-to: (?<url>.*)\]}

# This would result in:
regex.match(string)
# => #<MatchData "[image-link-to: https://www.devex.com] [no-caption]" url:"https://www.devex.com] [no-caption">
```
So our extracted URL would be `https://www.devex.com] [no-caption`.

What we can do instead is change the regex to either use the character class `[^\]]*` to include everything but the closing bracket:
```ruby
%r{\[image-link-to:\s+(?<url>[^\]]*)\]}
```
Or make the <url> part non-greedy using `.*?` instead of `.*`:
```ruby
%r{\[image-link-to:\s+(?<url>*?)\]}
```

It would be good to have this also covered by a unit test where our label contains another closing bracket and we expect the link to _not_ contain it.

Reply:
Oh, good catch! changin it.